### PR TITLE
Feedback

### PR DESCRIPTION
**Pull request by DaStanzel**
:wave:! GitHub Classroom created this pull request as a place for your teacher to leave feedback on your work. It will update automatically. **Don’t close or merge this pull request**, unless you’re instructed to do so by your teacher.

In this pull request, your teacher can leave comments and feedback on your code. Click the **Subscribe** button to be notified if that happens.

Click the **Files changed** or **Commits** tab to see all of the changes pushed to `master` since the assignment started. Your teacher can see this too.

<details>
<summary>
<strong>Notes for teachers</strong>
</summary>

Use this PR to leave feedback. Here are some tips:

- Click the **Files changed** tab to see all of the changes pushed to `master` since the assignment started. To leave comments on specific lines of code, put your cursor over a line of code and click the blue **+** (plus sign). To learn more about comments, read “[Commenting on a pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request)”.
- Click the **Commits** tab to see the commits pushed to `master`. Click a commit to see specific changes.
- If you turned on autograding, then click the **Checks** tab to see the results.
- This page is an overview. It shows commits, line comments, and general comments. You can leave a general comment below.

For more information about this pull request, read “[Leaving assignment feedback in GitHub](https://classroom.github.com/help/leaving-feedback-in-github)”.
</details>

Subscribed: @cr542

### CODE REVIEW

**Review by x21L — APPROVED** [review]

File: src/Calculator.java [hunk 1]
```diff
@@ -1,21 +1,142 @@
...
+
 public class Calculator {
 
-    // global variables
-    private static final Scanner scanner = new Scanner(System.in);
+	// global variables
+	private static final Scanner scanner = new Scanner(System.in);
+
+	private static double square(double a) {
+		return a * a;
```
Comment (x21L) [inline src/Calculator.java:9]: Inline return - great 😄

File: src/Calculator.java [hunk 2]
```diff
@@ -1,21 +1,142 @@
...
+//			Calculation algorithm throws an exception if the calculation cannot be made, for example if the users
+//			input is not a number. Exception will be handled here ...
+
+			try {
+				calculate(calcMethod, numberOne, numberTwo, choice);
+			} catch (NumberFormatException e) {
+				System.out.println("[*] Error: You have typed in invalid values for the calculation. "
+						+ "Make sure to type in numbers - nothing else is allowed!");
+				System.out.println("[*] Calculator is being terminated now. Please try again.");
+				return;
```
Comment (x21L) [inline src/Calculator.java:127]: You do not need this `return`. The catch block will stop the program anyhow.

File: src/Calculator.java [hunk 3]
```diff
@@ -1,21 +1,142 @@
...
+					+ Double.parseDouble(numberTwo) + "�) ... ");
+			System.out.println(
+					"[*] The result is: " + (pythagoras(Double.parseDouble(numberOne), Double.parseDouble(numberTwo))));
+			// Would be possible to calculate without using any own class methods using
+			// Math.sqrt().
+			// But hey, practice makes perfect :-)
+			break;
+		default:
+			System.out.println("[*] Your input is invalid. Calculator will be terminated.");
+			return;
```
Comment (x21L) [inline src/Calculator.java]: Why do you return from a void method?

File: src/Calculator.java [hunk 4]
```diff
@@ -1,21 +1,142 @@
...
+	private static double square(double a) {
+		return a * a;
+	}
+
+	public static double pythagoras(double a, double b) {
+		return Math.sqrt(square(a) + square(b));
+	}
+
+	public static void calculate(String calcMethod, String numberOne, String numberTwo, String choice)
+			throws NumberFormatException {
```
Comment (x21L) [inline src/Calculator.java:17]: Why do you throw an exception? For example, the `Integer.parseInt()` method does throw one anyways.

**Review by x21L — APPROVED** [review]
Some minor improvements can be made, like the unnecessary return statements. 
The solution with the switch and the if statements combined are very good. All in all a clean and readable style. Good work 😄

**Comment by ghost** [comment]
Many thanks for the good feedback :-)
I will make those corrections soon.

**Pull request merged**